python/tank_vendor/shotgun_authentication/user_impl.py

```diff
@@ -0,0 +1,360 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Defines the supported types of authentication methods with Shotgun. You can
+either authenticate with a session token with the SessionUser class or with an
+api key with the ScriptUser class. This module is meant to be used internally.
+"""
+
+import pickle
```

Comment:
I suggest if possible doing `import cPickle as pickle` for performance speedup. A tiny thing I know, but every little helps! :)